src/azureFunction/azureFunctionUtils.ts

```diff
@@ -203,7 +203,7 @@ export async function getAzureFunctionProjectFiles(): Promise<string[] | undefin
 		return undefined;
 	}
 	for (let host of hostFiles) {
-		let projectFile = await vscode.workspace.findFiles('*.csproj', path.dirname(host));
+		let projectFile = await vscode.workspace.findFiles('**/*.csproj', path.dirname(host));
```

Comment:
So this is supposed to be finding the csproj that are in the folder containing the host.json file, which in the issue filed seems to be the case (both are contained in the subfolder). And 

Was that not working or something?

Reply:
So I found that the issue is that even though we get the proper host file location, we do not get the proper csproj file associated with it based on the host location. This is an issue as now the find files properly search the host file directory.

I agree that the other issue in regards to both host.json and csproj being on the same level may not always be guaranteed. But as of now, I am not sure what else we could utilize that could strictly indicate that the folder is an azure function project.

Reply:
Oh, so the issue here is that the second parameter isn't the root folder to search from. 

https://github.com/microsoft/azuredatastudio/blob/main/src/vs/vscode.d.ts#L10652

`		export function findFiles(include: GlobPattern, exclude?: GlobPattern | null, maxResults?: number, token?: CancellationToken): Thenable<Uri[]>;`

```
		 * @param exclude  A {@link GlobPattern glob pattern} that defines files and folders to exclude. The glob pattern
		 * will be matched against the file paths of resulting matches relative to their workspace. When `undefined`, default excludes and the user's
		 * configured excludes will apply. When `null`, no excludes will apply.
```

So what we have there currently is incorrect and should be removed.

Instead you should look at using a [RelativePattern](https://github.com/microsoft/azuredatastudio/blob/main/src/vs/vscode.d.ts#L1932) and specify the root folder there.

Reply:
If it is at the root folder then it could possibly retrieve the wrong project file since right now we are specifying that a host.json and project file needs to be on the same level to ensure that it is an azure function. Right now the second parameter searches the given host.json file parent directory to see if there is also a csproj and if not then we show the user a warning.

Reply:
Hmm, this is a bit odd on how it's currently retrieving it correctly. But I will be sure to update, I see that the second parameter is supposed to be exclude. So should I be using the host.json parent folder (instead of the root folder) in the relative pattern instead to see if there is csproj located on the same level?